core/whistle-1.0.0/src/wh_cache.erl

```diff
@@ -143,6 +145,11 @@ start_link(Name, ExpirePeriod, Props) ->
                                    )
     end.
 
+-spec maybe_add_db_binding(wh_proplists()) -> wh_proplists().
+maybe_add_db_binding([[]]) -> [[]];
```

Comment:
sorry for the edit on previous comment.
we are not replacing, we are adding.
it will create
```
   [{conf,[federate]}]
   [{conf,[federate, {'type', <<"database">>}]}]
```